File: julia/anagrams.jl
function generatepermutations(a, n)
    if n == 1
        println(join(a, ""))
    else
        for i = 1:n-1
            generatepermutations(a, n-1)
            local j = iseven(n) ? i : 1
            a[j], a[n] = a[n], a[j]
        end
        generatepermutations(a, n-1)
    end
end

if length(ARGS) != 1
    println(stderr, "Exactly one argument is required")
    exit(1)
end
generatepermutations(split(ARGS[1], ""), length(ARGS[1]))

File: python/anagrams.py
import sys

def generatePermutations(a, n):
    if n == 0:
        print(''.join(a))
    else:
        for i in range(n):
            generatePermutations(a, n-1)
            j = 0 if n % 2 == 0 else i
            a[j], a[n] = a[n], a[j]
        generatePermutations(a, n-1)

if len(sys.argv) != 2:
    sys.stderr.write('Exactly one argument is required\n')
    sys.exit(1)
word = sys.argv[1]
generatePermutations(list(word), len(word)-1)
